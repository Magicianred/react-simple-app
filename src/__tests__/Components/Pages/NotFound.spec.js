import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { create } from 'react-test-renderer';

import NotFound from "./../../../Components/Pages/NotFound";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe("NotFound Component", () => {

    describe("Snapshots", () => {
    
        it("Matches the snapshot (with default title)", () => {
            const instance = create(<NotFound />);
            expect(instance.toJSON()).toMatchSnapshot();
        });
    
    });

    describe("Check content elements", () => {
        it("Render Component", () => {
            act(() => {
                render(<NotFound />, container);
            });
            expect(container.querySelector("h2").textContent).toBe("404 Page not found");
            expect(container.querySelector("p").textContent).toBe("The page is not found");
        });
    });
});