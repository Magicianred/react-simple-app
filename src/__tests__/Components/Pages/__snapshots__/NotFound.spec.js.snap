// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component Snapshots Matches the snapshot (with default title) 1`] = `
<div
  className="page"
>
  <div
    className="page notfound jumbotron"
  >
    <h2>
      404 Page not found
    </h2>
    <p>
      The page is not found
    </p>
  </div>
</div>
`;
